x-pack/plugins/timelines/public/components/actions/timeline/cases/add_to_new_case_button.tsx

```diff
@@ -25,22 +27,36 @@ const AddToNewCaseButtonComponent: React.FC<AddToNewCaseButtonProps> = ({
   owner,
   onClose,
 }) => {
-  const { addNewCaseClick, isDisabled, userCanCrud } = useAddToCase({
+  const { isDisabled, userCanCrud, caseAttachments, onCaseSuccess, onCaseCreated } = useAddToCase({
     event,
     useInsertTimeline,
     casePermissions,
     appId,
     owner,
     onClose,
   });
+  const { cases } = useKibana<TimelinesStartServices>().services;
```

Comment:
This change needs further work and clean up to come in other PRs. Effectively this pass the control from timelines having the flyout open/close logic to the CasesContext to handle it. See the usage of the new hook.

Reply:
This is the table you see when you a list of alerts in both security and observability

Reply:
can we just move the entire button out of timelines as a self contained component as was done in https://github.com/elastic/kibana/pull/114360/files